cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java

```diff
@@ -392,8 +392,10 @@ public void testBatchStatus() throws Exception {
     Assert.assertEquals(new NotFoundException(Id.Program.from("testnamespace1", "WordCountApp", ProgramType.FLOW,
                                                               "NotExist")).getMessage(),
                         returnedBody.get(0).get("error").getAsString());
-    Assert.assertEquals("'namespace:testnamespace1/application:WordCountApp/program:flows:NotExist' was not found.",
-                        returnedBody.get(0).get("error").getAsString());
+    // namespace:testnamespace1/application:WordCountApp/program:flows:NotExist
+    Assert.assertEquals(
+      String.format("'%s' was not found.", Id.Flow.from(TEST_NAMESPACE1, "WordCountApp", "NotExist").toString()),
```

Comment:
Asserting on message is always brittle. It might be enough to just assert on the exception class itself, right?